<a href="https://colab.research.google.com/github/Madhu678-coder/Assignment/blob/main/DataAi_Assignment_3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd
from sklearn.preprocessing import LabelEncoder, MinMaxScaler, StandardScaler
from sklearn.tree import DecisionTreeClassifier
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score

In [2]:
df = pd.read_csv("Social_Network_Ads.csv")

In [3]:
df.head()

User ID  Gender  Age  EstimatedSalary  Purchased
0  15624510    Male   19            19000          0
1  15810944    Male   35            20000          0
2  15668575  Female   26            43000          0
3  15603246  Female   27            57000          0
4  15804002    Male   19            76000          0

In [4]:
df.sample(5)

User ID  Gender  Age  EstimatedSalary  Purchased
116  15679297    Male   35            75000          0
363  15768293  Female   42            79000          0
393  15635893    Male   60            42000          1
133  15638963  Female   21            68000          0
122  15724423  Female   40            75000          0

In [5]:
df.isnull().sum()

User ID            0
Gender             0
Age                0
EstimatedSalary    0
Purchased          0
dtype: int64

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 400 entries, 0 to 399
Data columns (total 5 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   User ID          400 non-null    int64 
 1   Gender           400 non-null    object
 2   Age              400 non-null    int64 
 3   EstimatedSalary  400 non-null    int64 
 4   Purchased        400 non-null    int64 
dtypes: int64(4), object(1)
memory usage: 15.8+ KB


In [7]:
df.describe()

User ID         Age  EstimatedSalary   Purchased
count  4.000000e+02  400.000000       400.000000  400.000000
mean   1.569154e+07   37.655000     69742.500000    0.357500
std    7.165832e+04   10.482877     34096.960282    0.479864
min    1.556669e+07   18.000000     15000.000000    0.000000
25%    1.562676e+07   29.750000     43000.000000    0.000000
50%    1.569434e+07   37.000000     70000.000000    0.000000
75%    1.575036e+07   46.000000     88000.000000    1.000000
max    1.581524e+07   60.000000    150000.000000    1.000000

In [8]:
df_processed = df.drop("User ID", axis=1)
df_processed["Gender"] = LabelEncoder().fit_transform(df_processed["Gender"])

In [9]:
df_processed.corr()

Gender       Age  EstimatedSalary  Purchased
Gender           1.000000 -0.073741        -0.060435  -0.042469
Age             -0.073741  1.000000         0.155238   0.622454
EstimatedSalary -0.060435  0.155238         1.000000   0.362083
Purchased       -0.042469  0.622454         0.362083   1.000000

In [10]:
X = df_processed.drop("Purchased", axis=1)
y = df_processed["Purchased"]

In [22]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.25, random_state=0)

In [23]:
scaler_minmax = MinMaxScaler()
X_train_minmax = scaler_minmax.fit_transform(X_train)
X_test_minmax = scaler_minmax.transform(X_test)

In [24]:
scaler_standard = StandardScaler()
X_train_standard = scaler_standard.fit_transform(X_train)
X_test_standard = scaler_standard.transform(X_test)

In [31]:
clf_minmax = DecisionTreeClassifier(random_state=0)
clf_minmax.fit(X_train_minmax, y_train)
y_pred_minmax = clf_minmax.predict(X_test_minmax)
acc_minmax = accuracy_score(y_test, y_pred_minmax)
print(f"Accuracy with MinMaxScaler:, {acc_minmax:.4f}")

Accuracy with MinMaxScaler:, 0.9100


In [29]:
clf_standard = DecisionTreeClassifier(random_state=0)
clf_standard.fit(X_train_standard, y_train)
y_pred_standard = clf_standard.predict(X_test_standard)
acc_standard = accuracy_score(y_test, y_pred_standard)
print(f"Accuracy with StandardScaler:, {acc_standard:.4f}")

Accuracy with StandardScaler:, 0.9100


### THE ACCURACY WITH BOTH MINMAX_SCALER AND WITH STANDARDDCALER ARE EQUAL AND EQUAL TO 91%